src/data/polynomial/basic.lean

```diff
@@ -258,6 +258,9 @@ lemma C_pow : C (a ^ n) = C a ^ n := C.map_pow a n
 lemma C_eq_nat_cast (n : ℕ) : C (n : R) = (n : polynomial R) :=
 C.map_nat_cast n
 
+lemma mul_C_two (p : polynomial R) : 2 * p = (C 2) * p :=
```

Comment:
I'm not sure it's worth having this lemma. If anything, we would want a lemma `C 2 = 2`.

Reply:
We have that lemma right, it's `ring_hom.map_bit0`?

Reply:
You need to combine it with `map_one`.

Reply:
I like Johann's lemma, even if in my specific case it makes the proof slightly longer (I don't think I can tell `simp` to use `C 2 = 2`, if I do I get a deterministic timeout, probably because `simp` goes the other way).

Reply:
If you register simp lemmas
```
@[simp] lemma coeff_bit0_mul (P Q : polynomial ℤ) (n : ℕ) : 
  coeff (bit0 P * Q) n = bit0 (coeff (P * Q) n) :=
by simp [bit0, add_mul]

@[simp] lemma coeff_bit1_mul (P Q : polynomial ℤ) (n : ℕ) : 
  coeff ((bit1 P) * Q) n = bit0 (coeff (P * Q) n) + coeff Q n :=
by simp [bit1, add_mul]
```
(well, over any ring, not just `ℤ`), then your final simp concludes the proof without the need for `mul_C_two`.

Reply:
Thank you! Let's see if this works globally